README.md

```diff
@@ -39,18 +42,97 @@ let waiter = buildWaiter('friend-waiter');
 
 export default class Friendz extends Component {
   didInsertElement() {
-    waiter.beginAsync(this);
+    let token = waiter.beginAsync();
+
+    someAsyncWork()
+      .then(() => {
+        //... some work
+      })
+      .finally(() => {
+        waiter.endAsync(token);
+      });
+  }
+}
+```
+
+#### Custom Tokens
+
+The `beginAsync` method also can receive a custom token, if you need to specifically define a token to represent your async operation. _Note_ - due to the nature of
+async operations, it's impossible to guarantee which order the `endAsync` calls will run. If you are making multiple `beginAsync`/`endAsync` calls wiithin a single
+class or function, _you need to ensure the uniqueness of each async operation by ensuring the uniqueness of the tokens you provide_.
+
+The following code _will not_ guarantee uniqueness, as the two `endAsync` calls _may_ be called in succession:
+
+```js
+import Component from '@ember/component';
+import { buildWaiter } from 'ember-test-waiters';
+
+let waiter = buildWaiter('friend-waiter');
+
+export default class Friendz extends Component {
+  didInsertElement() {
+    let token = waiter.beginAsync(this);
 
     someAsyncWork().then(() => {
       waiter.endAsync(this);
     });
   }
+
+  action: {
+    someAction() {
+      let token = waiter.beginAsync(this);
+
+      someOtherAsync()
+        .then(() => {
+          //... some work
+        })
+        .finally(() => {
+          waiter.endAsync(this);
+        });
+    }
+  }
+}
+```
+
+The following code _will_ guarantee uniqueness, as the two `endAsync` calls _may_ be called in succession, but have unique tokens:
```

Comment:
I think we need to rephrase this stuff about "uniqueness", or at least explain what we mean before saying that the consumers should care about it. 

Frankly, there are absolutely times that you **do not** want uniqueness and instead you want to leverage our waiter system's assertion when two async things' ordering is not enforced.

Reply:
What are times when you'd not want uniqueness, for a discrete async operation?

Reply:
Removed custom waiter info from the README, for now.